src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,17 +1,55 @@
-import React from 'react';
+import React, { ChangeEvent } from 'react';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { actions } from '../../features/filter';
 
 export const TodoFilter: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const query = useAppSelector(state => state.query.query);
+
+  const handleQuery = (event: ChangeEvent<HTMLInputElement>) => {
+    dispatch(actions.setQuery(event.target.value));
+  };
+
+  const handleStatus = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    switch (event.target.value) {
+      case 'all':
+        dispatch(actions.setStatus('all'));
+        break;
+      case 'active':
+        dispatch(actions.setStatus('active'));
+        break;
+      case 'completed':
+        dispatch(actions.setStatus('completed'));
```

Comment:
- What is the point in the switch case here?
- all/active/completed are magic values